packages/commerce-sdk-react/src/components/ShopperExperience/Page/index.tsx

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2023, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import React, {useContext, useEffect, useState} from 'react'
+import {Page as PageType} from '../types'
+import {Region} from '../Region'
+
+type ComponentMap = {
+    [typeId: string]: React.ComponentType
+}
+
+interface PageProps extends React.ComponentProps<'div'> {
+    page: PageType
+    components: ComponentMap
+}
+
+type PageContextValue = {
+    components: ComponentMap
+}
+
+// This context will hold the component map as well as any other future context.
+export const PageContext = React.createContext({components: {}} as PageContextValue)
+
+// This hook allows sub-components to use the page context. In our case we use it
+// so that the generic <Component /> can use the component map to know which react component
+// to render.
+export const usePageContext = () => useContext(PageContext)
+
+/**
+ * This component will render a page designer page given its serialized data object.
+ *
+ * @param PageProps
+ * @returns JSX.Element
+ */
+export const Page = (props: PageProps) => {
+    const {page, components, className = '', ...rest} = props
```

Comment:
The `homepage-example` page response data includes props like:

```
 "pageKeywords": "meta keywords",
 "pageTitle": "page title",
  ...
  ```
  
  
  Is in the future Layout component where we plan to represent that data?

Reply:
Those SEO attributes are now supported in the page component (this was what I mentioned in stand up this morning).

These aren't related to layout components. But if for whatever reason we need to have access things like the `pageTitle` in a component, then we can think about adding it to the PageContextValue object.